lib/block-supports/layout.php

```diff
@@ -371,16 +371,24 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 		return (string) $content;
 	}
 
-	$block_gap              = gutenberg_get_global_settings( array( 'spacing', 'blockGap' ) );
-	$global_layout_settings = gutenberg_get_global_settings( array( 'layout' ) );
-	$has_block_gap_support  = isset( $block_gap ) ? null !== $block_gap : false;
-	$default_block_layout   = _wp_array_get( $block_type->supports, array( '__experimentalLayout', 'default' ), array() );
-	$used_layout            = isset( $block['attrs']['layout'] ) ? $block['attrs']['layout'] : $default_block_layout;
-
-	if ( isset( $used_layout['inherit'] ) && $used_layout['inherit'] ) {
-		if ( ! $global_layout_settings ) {
-			return $block_content;
-		}
+	static $static_information_computed   = false;
+	static $has_block_gap_support         = false;
+	static $global_layout_settings        = null;
+	static $root_padding_aware_alignments = false;
+	if ( ! $static_information_computed ) {
+		$global_settings               = gutenberg_get_global_settings();
+		$block_gap                     = _wp_array_get( $global_settings, array( 'spacing', 'blockGap' ), $global_settings );
+		$has_block_gap_support         = isset( $block_gap );
```

Comment:
My understanding is that block gap has three valid states: `null`, `true`, and `false`. This logic simplification is proper as per how the code is set up (`$block_gap` cannot be null if it is set), though I wonder what was the intention here. cc @ramonjd @andrewserong @aaronrobertshaw per thoughts as to how to update this code.

Reply:
I don't have a lot of background with the layout supports and block gap in particular, so can't speak much as to the original intention. 

The logic simplification in setting `$has_block_gap_support` make sense to me; however, I think the default value provided above when retrieving `$block_gap` is incorrect. According to the [docs](https://github.com/WordPress/gutenberg/blob/trunk/docs/how-to-guides/themes/theme-json.md#what-is-blockgap-and-how-can-i-use-it) it should default to `null`.

Reply:
@oandregal @aaronrobertshaw Also wondering about that. Unless there's a good answer why this was changed, I'd say we should revert this line to be similar to https://github.com/WordPress/gutenberg/pull/46074/files#diff-324697e41855298e2f2c74b078f174e0cbc9075cef736ce9c1e2c169bf64652eL376 again.

Reply:
> however, I think the default value provided above when retrieving $block_gap is incorrect. According to the [docs](https://github.com/WordPress/gutenberg/blob/trunk/docs/how-to-guides/themes/theme-json.md#what-is-blockgap-and-how-can-i-use-it) it should default to null.

Yes, the default of `blockGap` should be `null`. It looks like the defaults of the other `_wp_array_get` calls need to be updated too, I think.

Reply:
Hi all, I used global settings as the default because that's the default we were using before this change ( the function gutenberg_get_global_settings ends with _wp_array_get( $settings, $path, $settings ); so the default is the settings array if the path is not found). But I followed the feedback and applied the defaults suggested by @andrewserong.